jwql/instrument_monitors/miri_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,467 @@
+
+def get_cr_mags(jump_locs, jump_locs_pre, rateints, jump_data, jump_head, nints):
+    """
+        Computes a list of magnitudes using the coordinate of the detected jump compared to
+        the magnitude of the same pixel in the group prior to the jump.
+        
+        Parameters:
+        ----------
+        
+        jump_locs: list
+        List of coordinates to a pixel marked with a jump.
+        
+        jump_locs_pre: list
+        List of matching coordinates one group before jump_locs.
+        
+        head: FITS header
+        Header containing file information.
+        
+        rateints: ndarray
+        Array in DN/s.
+        
+        Returns:
+        -------
+        
+        mags: list
+        A list of cosmic ray magnitudes corresponding to each jump.
+        
+        """
+    
+    mags = []
+    for coord, coord_gb in zip(jump_locs, jump_locs_pre):
+        mags.append(magnitude(coord, coord_gb, rateints, jump_data, jump_head, nints))
+    
+    return mags
+
+def get_cr_rate(jump_locs, t):
+    """
+        Computes the rate of cosmic ray impacts in a given time.
+        
+        Parameters:
+        ----------
+        jump_locs: list
+        List of coordinates to a pixel marked with a jump.
+        
+        t: int or float
+        Time over which to compute the rate.
+        Nominally the effective exposure time.
+        
+        """
+    
+    return len(jump_locs)/t
+
+def get_jump_data(jump_filename):
+    """
+        Opens and reads a given .FITS file containing cosmic rays.
+        
+        Parameters:
+        ----------
+        jump_filename: str
+        Path to file.
+        
+        Returns:
+        -------
+        head: FITS header
+        Header containing file information
+        
+        data: NoneType
+        FITS data
+        
+        dq: ndarray
+        Data Quality array containing jump flags.
+        
+        """
+    
+    hdu = fits.open(jump_filename)
+    
+    head = hdu[0].header
+    data = hdu[1].data
+    
+    dq = hdu[3].data
+    
+    hdu.close()
+    
+    return head, data, dq
+
+def get_jump_locs(dq,nints):
+    """
+        Uses the data quality array to find the location of all jumps in the data.
+        
+        Parameters:
+        ----------
+        dq: ndarray
+        Data Quality array containing jump flags.
+        
+        Returns:
+        -------
+        jump_locs: list
+        List of coordinates to a pixel marked with a jump.
+        """
+    
+    temp = np.where(dq==4)
+    
+    jump_locs = []
+    
+    if nints > 1:
+        for i in range(len(temp[0])):
+            jump_locs.append((temp[0][i],temp[1][i],temp[2][i],temp[3][i]))
+    else:
+        for i in range(len(temp[0])):
+            jump_locs.append((temp[0][i],temp[1][i],temp[2][i]))
+
+return jump_locs
```

Comment:
I think this needs to be indented

Reply:
i believe this has to do with the weird indenting from before.